### ISSUE TITLE
(Another) request/suggestion regarding the 'Fn' key -- please do have a look.

### ISSUE THREAD

**Issue by ghost** [jigish/slate#262]
[I know there have been quite a few 'Fn' key related issues, and they all had to be closed because of problems making the feature work. However I had something to add, and wasn't sure if I would be able to get your attention if I posted to an already closed issue. 
So, I am creating a new issue. Feel free to delete/close this, if it makes sense]

I love Slate. But with some many configurations possible, one soon starts to run out of _convenient_ hotkeys that don't conflict with global ones or those used by other apps. 

And so, allowing Fn as a modifier (with all the combinations possible by combining it with other modifiers), would really be super helpful. And, at the current point, it might be _the_ single-most important feature that enables users to  make Slate an effortless part of their window management workflow -- for any and all types of window management tasks.

FWIW, using KeyRemap4MacBook allows one to bind to the Fn+key or Fn+modifier(s)+key type shortcuts. Even on Mountain Lion. Which means that it should be possible, in theory, for Slate to be able to allow Fn based based bindings as well.

**Comment by SethMilliken** [comment]
The core functionality of KeyRemap4MacBook is provided as a kernel extension. This presents a vastly different model of installation and authorization than Slate: Launching the Slate app vs launching the KeyRemap4MacBook installer, providing admin credentials, going through the installation procedure, and then rebooting (after which you are at risk of provoking kernel panics). Don't get me wrong, KeyRemap4MacBook is great; but it's a different class of tool. I doubt it would be worthwhile for Slate to make the same sort of compromises. Plus, there's nothing stopping you from using them together.

FWIW, using Slate's `modal-key`s can help greatly with available bindings.

**Comment by ghost** [comment]
@SethMilliken: Agree with you for the most part. However, KeyRemap4MacBook requires admin privileges only once, during installation. So as far as inconvenience goes (ignoring security aspects), its hardly an issue. Also, it might be worth investigating if admin privileges are required specifically for the Fn functionality (though I have a hunch it they would be, as well).
Using modal-key is a great option, I agree. I did not know about it, so thanks! But its not as seamless as an action that did not require switch modes etc.
Regarding using KeyRemap with Slate: Not a bad idea (I had thought earlier of doing that myself), but it still requires hotkeys to first be created using Slate, and mapping other (Fn-based) ones to those using KeyRemap. And these Slate hotkeys could still conflict/potentially conflict with other hotkeys, unless one choses really random/weird combinations that are unlikely to be used by the system/other app... but that's far from an ideal solution as far as I am concerned.

Having said that, I agree that you make useful points. If gigish/anyone else authorized to do so feels that the issue should be closed, they can go ahead and do that.
